Automattic/vip-cli
# WIP Add/dev env easy wp selection and image

## Description

This is still a Work in Progress. It needs unit tests and an accompanying branch in vip-container-images.

Adds dynamic selection of a WordPress Core repo based on the content/tags of the official WP git repo.

The list of tags gets filtered by a function called `collateTagList` which will generate a curated list of WP Core versions based on the following algorithm:

1. Sort the tag list descending from most recent to oldest.
2. Show all releases of the current version ( e.g. 5.8.2, 5.8.1, 5.8 )
3. Show only the newest releases of previous versions.
4. Only show a total of 20 entries.

![Screen Shot 2021-11-22 at 9 27 59 AM](https://user-images.githubusercontent.com/46167019/142899078-8f23aad2-4d48-4764-8a1f-c60102066cc6.png)


## Steps to Test

1. Check out PR.
1. update dependencies `npm install`
1. Run `npm run build`
1. Run `vip dev-env create`
